components/kyma-environment-broker/internal/ias/client_test.go

```diff
@@ -488,6 +539,33 @@ func (s *server) deleteSP(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+func (s *server) deleteSecrets(w http.ResponseWriter, r *http.Request) {
+	var deleteSecrets DeleteSecrets
+	err := json.NewDecoder(r.Body).Decode(&deleteSecrets)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		s.t.Errorf("test server cannot decode request body: %s", err)
+		return
+	}
+
+	if deleteSecrets.ClientID != userForRest {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	for _, sID := range deleteSecrets.ClientSecretsIDs {
+		for index, secret := range s.secrets {
+			if secret.ClientID == sID {
+				s.secrets[index] = s.secrets[len(s.secrets)-1]
+				s.secrets[len(s.secrets)-1] = ServiceProviderSecret{}
```

Comment:
I think you can use `s.secrets[index]` here or just remove above line and use `s.secrets[len(s.secrets)-1]`